security/pkg/k8s/controller/workloadsecret.go

```diff
@@ -488,27 +557,29 @@ func (sc *SecretController) scrtUpdated(oldObj, newObj interface{}) {
 
 	_, waitErr := sc.certUtil.GetWaitTime(scrt.Data[CertChainID], time.Now(), sc.minGracePeriod)
 
-	caCert, _, _, rootCertificate := sc.ca.GetCAKeyCertBundle().GetAllPem()
+	rootsWithExtraTrustAnchors := sc.getTrustAnchorsBundle()
```

Comment:
I think this is very expensive. We could:
1. Have a timestamp for the bundle, and in getTrustAnchorsBundle(), trigger an update if the last timestamp has passed a period, say 1 min or so.
2. Implement an listener for the bundle and update it when it's changed.
We need to think about what happens if multiple Citadels are updating the WL certs at the same time. What're the corner cases, for example, apiserver is temporarily unavailable for updating the bundle.

Reply:
I'll added a periodic timer check (option 1).

I would expect things to converge if multiple Citadel are updating certs while trust anchors are added/removed and the root CA is rotated.